src/PlaneWaveBasis.jl

```diff
@@ -140,24 +150,17 @@ end
 build_kpoints(basis::PlaneWaveBasis, kcoords) =
     build_kpoints(basis.model, basis.fft_size, kcoords, basis.Ecut)
 
-# This is the "internal" constructor; the higher-level one below should be preferred
+# Lowest-level constructor. All given parameters must be the same on all processors
+# and are stored in PlaneWaveBasis for easy reconstruction
 @timing function PlaneWaveBasis(model::Model{T}, Ecut::Number,
-                                kcoords::AbstractVector, ksymops, symmetries=nothing;
-                                fft_size=nothing, variational=true,
-                                optimize_fft_size=false, supersampling=2,
-                                kgrid=nothing, kshift=nothing,
-                                comm_kpts=MPI.COMM_WORLD,
-                               ) where {T <: Real}
-    mpi_ensure_initialized()
-
-    # Validate kpoints and symmetries
-    @assert length(kcoords) == length(ksymops)
-    if symmetries === nothing
-        # TODO instead compute the group generated by with ksymops, or
-        # just retire this constructor. Not critical because this
-        # should not be used in this context anyway...
-        symmetries = vcat(ksymops...)
+                                kcoords::AbstractVector, ksymops,
+                                fft_size, variational, symmetries,
+                                kgrid, kshift, comm_kpts) where {T <: Real}
+    if !(all(fft_size .== next_working_fft_size.(T, fft_size)))
+        error("Selected fft_size will not work for the buggy generic FFT routines")
```

Comment:
This error message does not help a user that has never encountered this error and does not know about the internals.

Reply:
Yeah I thought about that but probably if you want to start playing around with stuff like that you should be able to look at the code surrounding an error message; if not, you'd better turn back anyway

Reply:
You could at least point out the name of the cryptic function to use for fixing this.

Reply:
it's right there up one line from this error message. Looking up the stacktrace isn't particularly hard to do, and I don't think it's reasonable to babysit users who can't do that for this kind of advanced functionality

Reply:
You really don't want to be nice, do you? I would consider my self an advanced user and I hate looking up what the function name was when I simply forgot to call the function. If you print it, however, I don't have to actively dig through the code next time I use this feature. I would appreciate that. I assume others, too. ;)

Reply:
alright, alright. It's just so ugly that we even have this that I didn't want to enshrine it in an error message, but fair enough

Reply:
Yeah, but I doubt there will be a generic pure-Julia FFT any time soon. Attention pops up periodically, but then it decays when people find out how much effort went into FFTW.

Reply:
https://github.com/JuliaApproximation/FastTransforms.jl/blob/master/src/fftBigFloat.jl looks OK; doing a simple radix-2 FFT is not hard. It's probably not super fast, but we don't really care about that

Reply:
I know about that code. We tried initially and then moved away from it. Don't remember why, though. Could be that the one we use now is faster, actually.